src/extension.ts

```diff
@@ -109,12 +108,24 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 
         traceError(
             'Python interpreter missing:\r\n' +
-                '[Option 1] Select python interpreter using the ms-python.python.\r\n' +
-                `[Option 2] Set an interpreter using "${serverId}.interpreter" setting.\r\n` +
-                'Please use Python 3.8 or greater.',
+            '[Option 1] Select python interpreter using the ms-python.python.\r\n' +
+            `[Option 2] Set an interpreter using "${serverId}.interpreter" setting.\r\n` +
+            'Please use Python 3.8 or greater.',
         );
     };
 
+    let projectMetadata: Map<string, any> | undefined;
+    let heapUserId: string = '';
+    projectMetadata = context.globalState.get(PROJECT_METADATA);
+    if (projectMetadata) {
+        heapUserId = projectMetadata.get("username");
+    }
+
+    const sendHeapEventWithMetadata = async (eventName: string): Promise<void> => {
+        sendHeapEvent(eventName, projectMetadata, heapUserId);
+
+    }
```

Comment:
Can we move this code to `/common/telemetry` same file as `sendHeapEvent` ?

Reply:
@jitu5 How can we do this? The `projectMetadata` is only available within the context.

Reply:
@noklam 

We can keep it in separate function but need to pass `context` for every call

```
await sendHeapEventWithMetadata(CMD_RESTART_SERVER, context);
```  

```
export async function sendHeapEventWithMetadata(eventName: string, context: ExtensionContext): Promise<void> {
    let projectMetadata: Map<string, any> | undefined;
    let heapUserId: string = '';

    projectMetadata = context.globalState.get(PROJECT_METADATA);

    if (projectMetadata) {
        heapUserId = projectMetadata.get('username');
    }

    sendHeapEvent(eventName, projectMetadata, heapUserId);
}
```